MFLT-5975 Document hardware version for MCU projects (#222)
memfault/memfault-docs: docs/platform/software-version-hardware-version.mdx

(i.e. `smart-toaster` and `smart-oven`) and different board revisions of the
same device type (i.e. `smart-toaster-evt`, `smart-toaster-pvt`, etc.).

### MCU

> The software running on your devices will need to report its Hardware Version
> correctly such that it matches with what has been configured in the Project.
> Please see the
> [core/device info component](https://github.com/memfault/memfault-firmware-sdk/blob/master/components/include/memfault/core/platform/device_info.h)
> of the Firmware SDK for details on how to set this up on the device's software
> side.

Reviewer: For Android we derive a hardware version using different properties but for MCU there's no such requirement so I think we can just remove the "such that it matches with what has been configured in the Project."

```suggestion
The software running on your devices will need to report a Hardware Version
Please see the
[core/device info component](https://github.com/memfault/memfault-firmware-sdk/blob/master/components/include/memfault/core/platform/device_info.h)
of the Firmware SDK for details on how to set this up on the device's software
side.
```

Author: Good point, thx!
